File: aoc2019/aoc2019-java/src/test/java/de/havox_design/aoc2019/day10/Day10Test.java
package de.havox_design.aoc2019.day10;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.Arguments;
import org.junit.jupiter.params.provider.MethodSource;

import java.util.stream.Stream;

class Day10Test {

    @ParameterizedTest
    @MethodSource("getDataForTask1")
    void testTask1(String fileName, long expectation) {
        Assertions.assertEquals(expectation, Day10.processTask1(fileName));
    }

    private static Stream<Arguments> getDataForTask1() {
        return Stream.of(
                Arguments.of("de/havox_design/aoc2019/day10/day10part1sample1.txt", 8L),
                Arguments.of("de/havox_design/aoc2019/day10/day10part1sample2.txt", 33L),
                Arguments.of("de/havox_design/aoc2019/day10/day10part1sample3.txt", 35L),
                Arguments.of("de/havox_design/aoc2019/day10/day10part1sample4.txt", 41L),
                Arguments.of("de/havox_design/aoc2019/day10/day10part1sample5.txt", 210L)
        );
    }

    @ParameterizedTest
    @MethodSource("getDataForTask2")
    void testTask2(String fileName, long expectation) {
        Assertions.assertEquals(expectation, Day10.processTask2(fileName));
    }

    private static Stream<Arguments> getDataForTask2() {
        return Stream.of(
                Arguments.of("de/havox_design/aoc2019/day10/day10part2sample.txt", 0L)
        );
    }
}

File: aoc2023/src/test/kotlin/de/havox_design/aoc2023/day10/Day10Test.kt
package de.havox_design.aoc2023.day10

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.Arguments
import org.junit.jupiter.params.provider.MethodSource
import java.util.stream.Stream

class Day10Test {
    @Test
    fun testMainClass() {
        MainClass.main(arrayOf())
    }

    @ParameterizedTest
    @MethodSource("getDataForTestSolvePart1")
    fun testSolvePart1(filename: String, expectedResult: Long) =
        PipeMaze(filename).solvePart1().shouldBe(expectedResult)

    @ParameterizedTest
    @MethodSource("getDataForTestSolvePart2")
    fun testSolvePart2(filename: String, expectedResult: Long) =
        PipeMaze(filename).solvePart2().shouldBe(expectedResult)

    companion object {
        @JvmStatic
        private fun getDataForTestSolvePart1(): Stream<Arguments> =
            Stream.of(
                Arguments.of("day10Part1sample1.txt", 4L),
                Arguments.of("day10Part1sample2.txt", 8L)
            )

        @JvmStatic
        private fun getDataForTestSolvePart2(): Stream<Arguments> =
            Stream.of(
                Arguments.of("day10Part2sample1.txt", 4L),
                Arguments.of("day10Part2sample2.txt", 4L),
                Arguments.of("day10Part2sample3.txt", 8L),
                Arguments.of("day10Part2sample4.txt", 10L)
            )
    }
}

private fun Long.shouldBe(expectation: Long) = assertEquals(expectation, this)
